File: Python/TeemoAttacking/TeemoAttacking.py
class Solution(object):
    # (88ms)
	# 时间复杂度为O(n), 空间复杂度为O(1)
    def findPoisonedDuration(self, timeSeries, duration):
        """
        :type timeSeries: List[int]
        :type duration: int
        :rtype: int
        """
        if not timeSeries:
            return 0
        left, total = timeSeries[0], duration
        for time in timeSeries:
            if time < left + duration:
                total += time - left
            else:
                total += duration
            left = time
        return total


print Solution().findPoisonedDuration([1, 4], 2)
print Solution().findPoisonedDuration([1, 2], 2)

File: JavaScript/TeemoAttacking/TeemoAttacking.js
/**
 * @param {number[]} timeSeries
 * @param {number} duration
 * @return {number}
 */
// (109ms)
var findPoisonedDuration = function(timeSeries, duration) {
    if(timeSeries.length === 0) {
        return 0;
    }
    var left = timeSeries[0];
	var total = duration;
	timeSeries.forEach(function(item){
		total += item < left + duration ? item - left : duration;
		left = item;
	});
	return total;
};